github-flow.rst

```diff
@@ -0,0 +1,153 @@
+publikującego i to on podejmuje decyzje o:
+
+-  wewnętrznym procesie redakcyjnym w organizacji (np. upewnienie się,
+   że tekst nie zawiera literówek, jest przystępny dla osób postronnych,
+   w szczególności nietechnicznych)
+-  weryfikacji zasadności wkładu i jego wartości dla projektu (tzw.
+   sanity check)
+-  zakresie i formie wkładu (niesprzecznej z wcześniejszymi ustaleniami,
+   np. dokumenty publikowane są w formacie ``.rst``)
+
+Narzucenie nadmiernego procesu weryfikacji wewnętrznej może negatywnie
+wpłynąć na poczucie odpowiedzialności za wkład ("czy to rzeczywiście
+coś, co chcę opublikować pod swoim nazwiskiem"), a w konsekwencji może
+spowodować rozmycie odpowiedzialności w długim okresie.
+
+W skrócie: szanujemy nawzajem swój wkład w projekt i zakładamy, że
+publikowane są one zgodnie z najlepszą wiedzą publikującego.
+
+Proponujemy rozwiązania
+-----------------------
+
+GitHub działa na zasadzie tzw. pull requestów, które są pewną propozycją
+rozwiązania istniejącego problemu lub dodaniem nowej wartości do
+projektu.
+
+Issue
+~~~~~
+
+Na GitHubie narzędziem do śledzenia problemów i proponowanych usprawnień
+w projekcie są issues. Dzięki możliwości zgłaszania ich przez każdą
+osobę, która angażuje się w projekt open-source (m.in. deweloperów,
+użytkowników), pozwala na usprawnianie komunikacji i rozmawianie o
+priorytetach.
+
+Dla zachowania pełnej historii komunikacji warto jest dodać do
+repozytorium issue, które opisuje obszar problemowy i pozwala na
+dyskutowanie o możliwych rozwiązaniach. Każdy issue można następnie
+powiązać z konkretnym pull requestem, który go rozwiązuje.
+
+Publikowanie pull requesta
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Aby móc opublikować pull requesta należy:
+
+-  stworzyć forka repozytorium głównego na swoim koncie GH
+-  zrobić ``git pull`` na swoim forku i dodać go jako ``remote`` do repozytorium
+   lokalnego (zwyczajowo ``origin``)
+-  dodać główne repozytorium projektu jako ``remote`` (zwyczajowo ``upstream``)
+-  stworzyć brancha na proponowane rozwiązanie, dodawać na nim lokalnie
+   zmiany
+-  w momencie, kiedy zmiany są zadowalające dla publikującego, uporządkować
+   ``git log``
+-  poszczególne commity powinny być implementacją pełnych rozwiązań
+   (pozwala to np. na wybieranie tylko części commitów z pull requesta
+   [``cherrypicking``] i łatwe usunięcie niedziałającego rozwiązania w
+   przypadku ewentualnych problemów)
+-  zrobić ``git rebase`` proponowanych rozwiązań na najnowszą wersję
+   master brancha na ``upstream``
+-  to osoba proponująca rozwiązanie odpowiada za jego działanie i upewnienie
+   się, że współgra z istniejącym kodem
+-  zrobić ``git push`` na ``remote origin`` i opublikować pull reqesta do
+   głównego repozytorium
+
+Informacje w PR
+^^^^^^^^^^^^^^^
+
+Pull request powinien zawierać:
+
+-  krótką nazwę opisującą istotę rozwiązania (np. 'Implementacja autoryzacji
+   z OpenID'),
+-  tag dotyczący obszaru problemowego (np. 'uprawnienia'),
```

Comment:
Skoro znacznik trafia do pull requesta, a nie trafia (słusznie chyba) do commita, to czy nie lepiej zamiast znaczników stosować Githubowe etykiety?